alert_system/etl/base/extraction.py

```diff
@@ -0,0 +1,297 @@
+            "correlation_id": self._get_correlation_id(feature),
+            "resp_data": feature,
+            "connector": self.connector,
+        }
+
+    def get_datetime_filter(self) -> str:
+        """
+        Generate datetime filter string for STAC queries.
+
+        Returns:
+            ISO 8601 datetime range string
+        """
+        now = datetime.now(timezone.utc)
+        last_run = self.connector.last_success_run
+
+        start_time = last_run if last_run else (now - timedelta(days=10))  # NOTE: Arbitrary value for failure case.
+        return f"{start_time.isoformat()}/{now.isoformat()}"
+
+    def _build_filter(self, correlation_id: str, base_filter: Optional[Dict] = {}) -> Dict:
+        """Build filter dict with correlation ID."""
+        filters = base_filter.copy() if base_filter else {}
+        filters["filter"] = f"monty:corr_id = '{correlation_id}'"
+        return filters
+
+    def _fetch_items(self, endpoint: str | None, filter: Optional[Dict], correlation_id: str) -> Generator[Dict, None, None]:
+        """
+        Generic method to fetch items with correlation ID filtering.
+
+        """
+        url = f"{self.base_url}{endpoint}"
+        filters = self._build_filter(correlation_id=correlation_id, base_filter=filter)
+
+        return self.fetch_stac_data(url, filters)
+
+    def _save_stac_item(self, stac_id: str, defaults: Dict, item_type: str) -> Optional[ExtractionItem]:
+        """
+        Generic method to save or update STAC items.
+
+        """
+        try:
+            obj, created = ExtractionItem.objects.update_or_create(stac_id=stac_id, defaults=defaults)
+            action = "Created" if created else "Updated"
+            logger.info(f"{action} {item_type} {stac_id}")
+            return obj
+        except Exception as e:
+            logger.warning(f"Failed to save {item_type} {stac_id}: {e}", exc_info=True)
+            return None
+
+    def _extract_impact_items(self, stac_obj: ExtractionItem) -> List[ExtractionItem]:
+        """Process impact items related to a STAC event object."""
+        if not self.impact_endpoint:
+            logger.info("No impact endpoint defined.")
+            return []
+        impact_objects: List[ExtractionItem] = []
+        try:
+            impact_features = self._fetch_items(self.impact_endpoint, self.filter_impact, stac_obj.correlation_id)
+        except Exception as e:
+            logger.warning(f"Failed to fetch impacts for event {stac_obj.stac_id}: {e}")
+            return []
+
+        for feature in impact_features:
+            impact_id = feature.get("id", None)
+            if not impact_id:
+                logger.warning(f"Impact feature missing 'id': {feature}")
+                continue
+
+            defaults = self._build_base_defaults(feature)
+            defaults.update({"collection": ExtractionItem.CollectionType.IMPACT})
+            impact_object = self._save_stac_item(impact_id, defaults, "impact")
```

Comment:
Better to get the type from collection instead of passing the hardcoded type as argument.

```suggestion
            impact_object = self._save_stac_item(impact_id, defaults)
```